tests/execution/test-required-fields.md

```diff
@@ -0,0 +1,126 @@
+# Test API
+
+```graphql @config
+schema @server @upstream(baseURL: "http://jsonplaceholder.typicode.com") {
+  query: Query
+}
+
+type Query {
+  users: [User] @http(path: "/users")
+  posts: [Post] @http(path: "/posts")
```

Comment:
please, check the [spec](https://spec.graphql.org/October2021/#sec-Handling-Field-Errors).

Seems like, since the field `posts` is nullable it should be set to null instead of propagating up to the whole data